homeassistant/components/ecovacs/__init__.py

```diff
@@ -46,21 +47,22 @@
 )
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Ecovacs component."""
     _LOGGER.debug("Creating new Ecovacs component")
 
     hass.data[ECOVACS_DEVICES] = []
 
-    ecovacs_api = EcoVacsAPI(
+    ecovacs_api = await hass.async_add_executor_job(
```

Comment:
It looks like a few of these executor calls could be combined if you make a new function that runs in the executor